main.go

```diff
@@ -73,30 +82,38 @@ func main() {
 	r.Handler("GET", *metricsPath, prometheus.Handler())
 
 	// Handlers for pushing and deleting metrics.
-	r.PUT("/metrics/job/:job/*labels", handler.Push(ms, true))
-	r.POST("/metrics/job/:job/*labels", handler.Push(ms, false))
-	r.DELETE("/metrics/job/:job/*labels", handler.Delete(ms))
-	r.PUT("/metrics/job/:job", handler.Push(ms, true))
-	r.POST("/metrics/job/:job", handler.Push(ms, false))
-	r.DELETE("/metrics/job/:job", handler.Delete(ms))
+	pushAPIPath := *routePrefix + "/metrics"
+	r.PUT(pushAPIPath+"/job/:job/*labels", handler.Push(ms, true))
+	r.POST(pushAPIPath+"/job/:job/*labels", handler.Push(ms, false))
+	r.DELETE(pushAPIPath+"/job/:job/*labels", handler.Delete(ms))
+	r.PUT(pushAPIPath+"/job/:job", handler.Push(ms, true))
+	r.POST(pushAPIPath+"/job/:job", handler.Push(ms, false))
+	r.DELETE(pushAPIPath+"/job/:job", handler.Delete(ms))
 
 	// Handlers for the deprecated API.
-	r.PUT("/metrics/jobs/:job/instances/:instance", handler.LegacyPush(ms, true))
-	r.POST("/metrics/jobs/:job/instances/:instance", handler.LegacyPush(ms, false))
-	r.DELETE("/metrics/jobs/:job/instances/:instance", handler.LegacyDelete(ms))
-	r.PUT("/metrics/jobs/:job", handler.LegacyPush(ms, true))
-	r.POST("/metrics/jobs/:job", handler.LegacyPush(ms, false))
-	r.DELETE("/metrics/jobs/:job", handler.LegacyDelete(ms))
-
-	r.Handler("GET", "/static/*filepath", prometheus.InstrumentHandler(
+	r.PUT(pushAPIPath+"/jobs/:job/instances/:instance", handler.LegacyPush(ms, true))
+	r.POST(pushAPIPath+"/jobs/:job/instances/:instance", handler.LegacyPush(ms, false))
+	r.DELETE(pushAPIPath+"/jobs/:job/instances/:instance", handler.LegacyDelete(ms))
+	r.PUT(pushAPIPath+"/jobs/:job", handler.LegacyPush(ms, true))
+	r.POST(pushAPIPath+"/jobs/:job", handler.LegacyPush(ms, false))
+	r.DELETE(pushAPIPath+"/jobs/:job", handler.LegacyDelete(ms))
+
+	r.Handler("GET", *routePrefix+"/static/*filepath", prometheus.InstrumentHandler(
 		"static",
 		http.FileServer(
-			&assetfs.AssetFS{Asset: Asset, AssetDir: AssetDir, AssetInfo: AssetInfo},
+			&assetfs.AssetFS{
+				Asset: func(name string) ([]byte, error) {
+					// Note: this will do bad things if routePrefix is multi-byte,
+					// but I don't think that's valid in a URL anyway.
```

Comment:
That's actually fine. `len` on a string acts on bytes, not runes, and the slice operations also act on the byte level.

Reply:
🤔 I thought I understood this... I remember going back and forth on this very late at night and I can't remember now why I thought it could cause a problem 🤷‍♂️